cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/ResourceDistributionGoal.java

```diff
@@ -127,9 +174,19 @@ public ModelCompletenessRequirements clusterModelCompletenessRequirements() {
   protected boolean selfSatisfied(ClusterModel clusterModel, BalancingAction action) {
     Broker destinationBroker = clusterModel.broker(action.destinationBrokerId());
     Replica sourceReplica = clusterModel.broker(action.sourceBrokerId()).replica(action.topicPartition());
-    // If the source broker is dead and currently self healing dead brokers only, then the action must be executed.
+    // If the source broker is dead and currently self healing dead brokers only, unless it is replica swap, the action
+    // must be executed.
     if (!sourceReplica.broker().isAlive() && _selfHealingDeadBrokersOnly) {
-      return true;
+      return action.balancingAction() != REPLICA_SWAP;
+    }
+
+    if (action.balancingAction() == REPLICA_SWAP) {
```

Comment:
Can we use `switch` here as well? Or put the rest of the code in the `else` statement.